### PR TITLE
Handle string-based engine argument to `read_json`

### PR DESCRIPTION
**Pull request by rjzamora**
~**TODO**: Still haven't tested this locally with a pandas nightly~

- [x] Closes https://github.com/dask/dask/issues/9943
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/io/tests/test_json.py [hunk 1]
```diff
@@ -120,6 +121,26 @@ def _my_json_reader(*args, **kwargs):
         assert_eq(actual, actual_pd)
 
 
+@pytest.mark.parametrize(
+    "engine",
+    [
+        pytest.param(
+            "ujson",
+            marks=pytest.mark.skipif(
+                parse_version(pd.__version__) < parse_version("2.0"),
```
Comment (jrbourbeau) [inline dask/dataframe/io/tests/test_json.py:130]: We've got a compatibility variable for this

https://github.com/dask/dask/blob/0890b96be75e46ef2b88c11015761b9a6bd32480/dask/dataframe/_compat.py#L19

File: dask/dataframe/io/json.py [hunk 2]
```diff
@@ -209,6 +212,9 @@ def read_json(
     if path_converter is None:
         path_converter = lambda x: x
 
+    # Handle engine string (Pandas>=2.0)
```
Comment (jrbourbeau) [inline dask/dataframe/io/json.py:215]: It's probably worth raising an informative error if `engine=` is a string and an older version of `pandas` is being used

**Review by jrbourbeau — COMMENTED** [review]
Nice, thanks @rjzamora. Should we consider deprecating passing callables to `engine` in favor of the recent backend dispatching work? 

Also cc @phofl for visibility

File: dask/dataframe/io/json.py [hunk 3]
```diff
@@ -157,9 +158,11 @@ def read_json(
         Text conversion, ``see bytes.decode()``
     compression : string or None
         String like 'gzip' or 'xz'.
-    engine : function object, default ``pd.read_json``
+    engine : callable or str, default ``pd.read_json``
         The underlying function that dask will use to read JSON files. By
         default, this will be the pandas JSON reader (``pd.read_json``).
+        If a string is specified, this value will be passed under the ``engine``
+        key-word argument to ``pd.read_json`` (only supported for Pandas-2.0).
```
Comment (phofl) [inline dask/dataframe/io/json.py:165]: only supported starting from pandas-2.0?

Comment (rjzamora) [reply dask/dataframe/io/json.py:165]: >only supported starting from pandas-2.0?

Does `pd.read_json` accept an engine argument for `pandas<2.0`? If so, my mistake.

Comment (phofl) [reply dask/dataframe/io/json.py:165]: No, sorry. I meant rewording it. This sounds to me as it would only be accepted for 2.0 and not 2.1 for example

**Comment by rjzamora** [comment]
>Should we consider deprecating passing callables to engine in favor of the recent backend dispatching work?

The "cudf"-backend version of `read_json` is just calling the "pandas"-backend version with `engine=cudf.read_json`. Therefore, I'd prefer keep the callable option if possible (not entirely sure of a good alternative).

File: dask/dataframe/io/json.py (hunk 3 above)
Comment (rjzamora) [reply dask/dataframe/io/json.py:165]: >This sounds to me as it would only be accepted for 2.0 and not 2.1 for example

Ahah. good point!

**Comment by phofl** [comment]
@rjzamora do you have the bandwidth to continue? Can take over if not.

**Comment by rjzamora** [comment]
>@rjzamora do you have the bandwidth to continue? Can take over if not.

Sorry - This one fell off my radar! Hopefully just addressed the old review comments.

File: dask/dataframe/io/tests/test_json.py [hunk 4]
```diff
@@ -120,6 +121,26 @@ def _my_json_reader(*args, **kwargs):
         assert_eq(actual, actual_pd)
 
 
+@pytest.mark.parametrize(
+    "engine",
+    [
+        pytest.param(
+            "ujson",
+            marks=pytest.mark.skipif(
```
Comment (phofl) [inline dask/dataframe/io/tests/test_json.py:129]: Could you configure the test to check the error message if pandas < 2.0 instead of skipping it?

Comment (rjzamora) [reply dask/dataframe/io/tests/test_json.py]: Yeah, sorry about that!

**Review by jrbourbeau — APPROVED** [review]
Thanks @rjzamora for the changes here and @phofl  for reviewing. Will merge after CI finishes 

EDIT: Note that CI failures are unrelated to the changes here and have either already been resolved on `main` or are in the process of being resolved

**Pull request merged**